Userland/Applications/PixelPaint/VectorscopeWidget.cpp

```diff
@@ -124,11 +124,26 @@ void VectorscopeWidget::paint_event(GUI::PaintEvent& event)
-        auto graticule_rect = Gfx::FloatRect::centered_at(primary_color.to_vector(scope_size), { graticule_size, graticule_size }).to_rounded<int>();
-        base_painter.draw_rect_with_thickness(graticule_rect, graticule_color, graticule_thickness);
-        auto text_rect = graticule_rect.translated(graticule_size / 2, graticule_size / 2);
-        base_painter.draw_text(text_rect, StringView { &primary_color.symbol, 1 }, Gfx::TextAlignment::TopLeft, graticule_color);
+        auto center = primary_color.to_vector(scope_size);
+        auto center_rounded = primary_color.to_vector(scope_size).to_rounded<int>();
+        // Box color
+        Gfx::Color corner_color = Gfx::Color::from_yuv(0.5f, primary_color.u, primary_color.v).saturated_to(0.5);
+        // Top Left Corner
+        base_painter.draw_line(Gfx::IntPoint(center_rounded.x() - graticule_size / 2, center_rounded.y() - graticule_size / 2), Gfx::IntPoint(center_rounded.x() - graticule_size / 3, center_rounded.y() - graticule_size / 2), corner_color, graticule_thickness);
+        base_painter.draw_line(Gfx::IntPoint(center_rounded.x() - graticule_size / 2, center_rounded.y() - graticule_size / 2), Gfx::IntPoint(center_rounded.x() - graticule_size / 2, center_rounded.y() - graticule_size / 3), corner_color, graticule_thickness);
+        // Top Right Corner
+        base_painter.draw_line(Gfx::IntPoint(center_rounded.x() + graticule_size / 2, center_rounded.y() - graticule_size / 2), Gfx::IntPoint(center_rounded.x() + graticule_size / 3, center_rounded.y() - graticule_size / 2), corner_color, graticule_thickness);
+        base_painter.draw_line(Gfx::IntPoint(center_rounded.x() + graticule_size / 2, center_rounded.y() - graticule_size / 2), Gfx::IntPoint(center_rounded.x() + graticule_size / 2, center_rounded.y() - graticule_size / 3), corner_color, graticule_thickness);
+        // Bottom Left Corner
+        base_painter.draw_line(Gfx::IntPoint(center_rounded.x() - graticule_size / 2, center_rounded.y() + graticule_size / 2), Gfx::IntPoint(center_rounded.x() - graticule_size / 3, center_rounded.y() + graticule_size / 2), corner_color, graticule_thickness);
+        base_painter.draw_line(Gfx::IntPoint(center_rounded.x() - graticule_size / 2, center_rounded.y() + graticule_size / 2), Gfx::IntPoint(center_rounded.x() - graticule_size / 2, center_rounded.y() + graticule_size / 3), corner_color, graticule_thickness);
+        // Bottom Right Corner
+        base_painter.draw_line(Gfx::IntPoint(center_rounded.x() + graticule_size / 2, center_rounded.y() + graticule_size / 2), Gfx::IntPoint(center_rounded.x() + graticule_size / 3, center_rounded.y() + graticule_size / 2), corner_color, graticule_thickness);
+        base_painter.draw_line(Gfx::IntPoint(center_rounded.x() + graticule_size / 2, center_rounded.y() + graticule_size / 2), Gfx::IntPoint(center_rounded.x() + graticule_size / 2, center_rounded.y() + graticule_size / 3), corner_color, graticule_thickness);
```

Comment:
Can you extract the points to separate variables, as many points are used more than once?

Reply:
Extracted and updated